aten/src/ATen/native/AdaptiveAveragePooling3d.cpp

```diff
@@ -292,9 +295,23 @@ Tensor adaptive_avg_pool3d_cpu(Tensor const& input, IntArrayRef output_size) {
   return output;
 }
 
-Tensor& adaptive_avg_pool3d_backward_out_cpu(const Tensor& gradOutput_,
-    const Tensor& input,
-    Tensor& gradInput) {
+Tensor adaptive_avg_pool3d(at::Tensor const& input, IntArrayRef output_size) {
+  TORCH_CHECK(output_size.size() == 3, "adaptive_avg_pool3d: output_size must be 3");
+
+  if (output_size[0] == 1 && output_size[1] == 1 && output_size[2] == 1) {
+    // in this case, adaptive pooling is just computing mean over hw
+    // dimensions, which can be done more efficiently
+    Tensor out = input.mean({-1, -2, -3}, /* keepdim = */ true);
+    return out;
+  } else {
+    return _adaptive_avg_pool3d(input, output_size);
+  }
+}
+
+Tensor& adaptive_avg_pool3d_backward_out_cpu(
```

Comment:
looks like signature changed here, causing linking errors (should be `const Tensor& gradOutput_,
    const Tensor& input,
    Tensor& gradInput`)